src/gui/app.cpp

```diff
@@ -1,18 +1,150 @@
 #include "app.h"
 
+#include "arguments.h"
+#include "mainwindow.h"
+#include "printinfo.h"
+#include "version.h"
+
 #include <QDir>
 #include <QFileInfo>
 #include <QFileOpenEvent>
 #include <QSettings>
 #include <QTextStream>
-
-#include "arguments.h"
-#include "mainwindow.h"
-#include "printinfo.h"
-#include "version.h"
+#include <vector>
 
 namespace NeovimQt {
 
+MainWindow* s_lastActiveWindow{ nullptr };
+std::vector<MainWindow*> s_windows;
+int s_exitStatus{ 0 };
+
+struct ConnectorInitArgs
+{
+	enum class Type
+	{
+		Embed,
+		Server,
+		Spawn,
+		Default
+	};
+
+	ConnectorInitArgs(const QCommandLineParser& parser, QStringList nvimArgs)
+		: type{ getConnectorType(parser) }
+		, timeout{ parser.value("timeout").toInt() }
+		, server{ parser.value("server") }
+		, nvim{ parser.value("nvim") }
+		, positionalArgs{ parser.positionalArguments() }
+		, neovimArgs{ std::move(nvimArgs) }
+	{
+	}
+
+	ConnectorInitArgs(
+		Type _type, int _timeout, QString _server, QString _nvim, QStringList nvimArgs)
+		: type{ _type }
+		, timeout{ _timeout }
+		, server{ std::move(_server) }
+		, nvim{ std::move(_nvim) }
+		, positionalArgs{}
+		, neovimArgs{ std::move(nvimArgs) }
```

Comment:
Same here. Please double-check your move semantics.

Generally you should use:
1. `T&& val` with `m_var{ std::move(val) }`
2. `const T& val` with `m_var{ val }`

The move here doesn't do much, since your parameter is copied by value.

Based on looking at the caller, I think you want `T&&` as the type instead.

Reply:
I use this approach with simple types that I always will copy. There's no condition that changes the fact that I will not be using `_nvim`. This is pattern so that I don't have to have one ctor with the move only parameter and one that takes a const ref. Stephen Meyers has this in his book as well. 

Do you prefer that I get rid of this? Or create a new ctro that takes `T&&` and one that takes `const T&`?

Reply:
Sorry for the confusion. The `_varName` is fine.

I'm talking about the `std::move`. It doesn't do anything as written.

This will move the values:
```
..., QString&& _server, ...
...
, server{ std::move(_server) }
...
```

This code is probably not what you intended, but it is simpler and functionally equivalent to what you wrote:
```
..., const QString& _server, ...
...
, server{ _server }
...
```

Reply:
I have no issues with the `_name` pattern.

If I'm understanding your comment correctly... This does in fact `std::move` here. Right?

I prefer being explicit with about `T&&` to avoid accidental copy (you aren't actually using copy, only move here right?). If you do need the copy and move, feel free to keep as-is.

I've never seen this pattern before. I'll pull out my copy of Scott Meyers "Effective Modern C++" and read more.

Thanks for the clarification. Again... I learned something new today! :smile:

Reply:
> If I'm understanding your comment correctly... This does in fact std::move here. Right?

Yes, it indeed does a move.

> I prefer being explicit with about `T&&` to avoid accidental copy (you aren't actually using copy, only move here right?). If you do need the copy and move, feel free to keep as-is.

I prefer this as well, but in this case there's really not that big of a gain from having two different ctors one of which takes `T&&` and the other `const T&`. So, I'll keep it as it is. 

> Thanks for the clarification. Again... I learned something new today! 😄

Well, I'm glad the tables turned and I was able to be useful to you. :D